keadm/cmd/keadm/app/cmd/util/common.go

```diff
@@ -536,17 +559,32 @@ func IsKubeEdgeProcessRunning(proc string) (bool, error) {
 }
 
 func isEdgeCoreServiceRunning(serviceName string) (bool, error) {
```

Comment:
IMHO, i think if the edgecore is just enabled but not running(start failed due to some errors), we should also return true.
So i'm thinking how about abstract two functions: `isServiceRunning` and `isServiceEnabled`

Reply:
I have updated the change to have those two different functions and changed the check for docker running to use `isServiceEnabled`. Should we be checking that docker is enabled or actually running though?

Reply:
i think for docker, we can use both `isServiceRunning` and `isServiceEnabled`